### DISCUSSION TITLE
More accessible way to set the Texture Property on SkullMeta

### DISCUSSION THREAD

**Discussion by rowan-vr** [PaperMC/Paper#11457] [Ideas]
### Is your feature request related to a problem?

This feature request is related to the problem that it is -- as plugin developer -- unnecessarily difficult to set the texture of a player head to a signed texture, often I solve this using some reflection (which became more difficult in the later versions due to ResolvableProfile), but I think it'd be beneficial to all plugin to implement an more straight forward way to achieve this.

### Describe the solution you'd like.

I'd introduce this in either one of the following ways, I'd be happy to hear some feedback if there're other ideas to achieve this!

1. Introduce a 'setUrl' method on SkullMeta, under the hood this method works as follows:
	- It creates a property value of proper format from this URL for example a taco skull
```
"textures": {
      "SKIN": {
        "url": "http://textures.minecraft.net/texture/98ced74a22021a535f6bce21c8c632b273dc2d9552b71a38d57269b3538cf"
      }
}
```
```
eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvOThjZWQ3NGEyMjAyMWE1MzVmNmJjZTIxYzhjNjMyYjI3M2RjMmQ5NTUyYjcxYTM4ZDU3MjY5YjM1MzhjZiJ9fX0=
```

2. Introduce a setTextureProperty method using the PlayerProperty class to directly set this textures tag of the skull, under the hood it can create a resolvable profile from a mock profile with this property set

3. Introduce a (private) setProfile(GameProfile) method so plugin developers can set the profile of a skull without having to use paperweight and only mojang authlib, they would still be required access this method with reflection however.

### Describe alternatives you've considered.

Like listed above, an alternative would be to use lots of reflection and nowadays paperweight userdev to be able to create a resolvable profile. 

### Other

PS: if accepted I'd love to add this :D

**Comment by masmc05** [comment]
Use either one
```java
SkullMeta meta;
URI uri; // In the example is http://textures.minecraft.net/texture/98ced74a22021a535f6bce21c8c632b273dc2d9552b71a38d57269b3538cf
// Set both to actual values here
var profile = Bukkit.createProfile(new UUID(0, 0), "");
var textures = profile.getTextures();
textures.setSkin(uri.toURL());
profile.setTextures(textures);
meta.setPlayerProfile(profile);
```
or
```java
SkullMeta meta;
String base64; // In the example is eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvOThjZWQ3NGEyMjAyMWE1MzVmNmJjZTIxYzhjNjMyYjI3M2RjMmQ5NTUyYjcxYTM4ZDU3MjY5YjM1MzhjZiJ9fX0=
// Set both to actual values here
var profile = Bukkit.createProfile(new UUID(0, 0), "");
var property = new ProfileProperty("textures", base64);
// var property = new ProfileProperty("textures", base64, signature);
profile.setProperty(property);
meta.setPlayerProfile(profile);
```

**Reply by rowan-vr** [reply]
I did not know that bukkit.createprofile exists, thank you :D
